### ISSUE TITLE
Auto-Type missing in the Linux snap build of version 2.7.0

### ISSUE THREAD

**Issue by paddylandau** [keepassxreboot/keepassxc#7567]
## Overview

In the snap build of version 2.7.0, the Auto-Type functionality is entirely missing.

## Steps to Reproduce

1. `snap install keepassxc` (or `snap refresh` if the previous version is already installed)
1. `snap run keepassxc`

## Expected Behavior

- Auto-Type functionality is shown in the Settings.
- It works when using it (e.g. Ctrl+Shift+V or Ctrl+Alt+A).
- It is available in the right-click menu for an entry.

## Actual Behavior

- Auto-Type functionality is missing from the Settings.
- Ctrl+Shift+V and Ctrl+Alt+A don't work.
- It isn't available on the right-click menu for an entry (see attached screenshot).

When running from a terminal, the following error messages are displayed. (The two YubiKey messages are probably irrelevant, as they also occur with the PPA version. I don't use YubiKey.)

    $ snap run keepassxc
    Qt: Session management error: Could not open network socket
    YubiKey: Failed to establish PCSC context.
    YubiKey: PCSC interface is disabled or not initialized.
      OpenType support missing for "Saab", script 13

## Other notes

The [official PPA version](https://launchpad.net/~phoerious/+archive/ubuntu/keepassxc) of KeePassXC works correctly.

### Help → About → Debug Info

KeePassXC - Version 2.7.0
Revision: d7a9ef4
Distribution: Snap

Qt 5.15.3
Debugging mode is disabled.

Operating system: Ubuntu Core 20
CPU architecture: x86_64
Kernel: linux 5.13.0-37-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.12.1

### Environment 

- Operating System: Linux Ubuntu 20.04.4 LTS 64-bit
- Desktop Env: Gnome 3.36.8
- Windowing System: X11

![KeePassXC missing Auto-Type functionality](https://user-images.githubusercontent.com/16425112/159448504-90f5ec23-9497-489c-b26c-e404a34e36c5.png)

**Comment by droidmonkey** [comment]
I fixed it and pushed the new snap build to stable channel. You can get the latest with a `sudo snap refresh keepassxc`. Will be baked into 2.7.1.

**Comment by fbau3r** [comment]
Works for me again! (and didn't work before like @paddylandau described)

You wouldn't believe how much this feature was missed throughout this whole day... ;-)
Thank you very much for the fast response and fix!

**Comment by paddylandau** [comment]
@droidmonkey — Thank you for fixing this so fast.

I can see in the updated KeePassXC that the functionality is back, e.g. in the right-click menu.

But it just doesn't work for me. When I use, for example, Ctrl+Shift+V, or right-click > Perform Auto-Type, nothing happens!

If the snap works for @fbau3r, then it must be a problem with my system. So, don't worry; I'll just use the PPA instead, because that's working for me.

Thank you again for a speedy response!

**Comment by fbau3r** [comment]
@paddylandau My Global Shortcut Keybinding was not assigned anymore in settings. I reassigned it, then it worked again.

**Comment by droidmonkey** [comment]
@paddylandau that may be because of a Wayland/X11 problem, hard to tell.

**Comment by Rayv1** [comment]
Same for me. Its not working since 2.7.0
I Press Ctrl+Shift+V and the dialog "Confirm Auto-Type" appears. After confirming keepassxc moves to background but nothing happens. 

-
KeePassXC - Version 2.7.0
Revision: 20c9ac1
Distribution: Snap

Qt 5.15.3
Debugging mode is disabled.

Operating system: Ubuntu Core 20
CPU architecture: x86_64
Kernel: linux 5.13.0-37-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.12.1

**Comment by Rayv1** [comment]
Okay i "fixed" it. In my Case under Settings there was enabled
- Use entry title to match windows for global Auto-Type
- Use entry URL to match windows for global Auto-Type
- Always ask before performing Auto-Type
seems like this is enabled by default since 2.7.0 (or is completely new)
i disabled all points and it works like a charm now.

**Comment by droidmonkey** [comment]
That's a different bug. It's supposed to work with the warning dialog but it was broken. Already fixed for the next update.

**Comment by piattj** [comment]
Sorry to harp on but I get the same issue with 2.7.0 on macOS. Same issue on two macs. 
Disabling these makes no difference.
Use entry title to match windows for global Auto-Type
Use entry URL to match windows for global Auto-Type
Always ask before performing Auto-Type

**Comment by droidmonkey** [comment]
Is fixed already thanks

**Comment by piattj** [comment]
> 
OK thanks. I presume in 2.7.1? Is there a downloadable 2.7.1. for macOS?

**Comment by droidmonkey** [comment]
Https://snapshot.keepassxc.org

**Comment by piattj** [comment]
> [Https://snapshot.keepassxc.org](Https://snapshot.keepassxc.org)

Thanks... just to check - is that the Apple Silicon (ARM) version, or Intel? Ta...

**Comment by phoerious** [comment]
We don't build Apple Silicon snapshots. Use the x64 version.

**Comment by jmper** [comment]
When trying to sudo snap refresh keepassxc I get "snap "keepassxc" has no updates available". The official snap package is still shown as 2.7.0 (on https://keepassxc.org/download/). Where can I get 2.7.1?

**Comment by droidmonkey** [comment]
When it's released

**Comment by washiur17** [comment]
Waiting for it.

**Comment by bastiencyr** [comment]
Hello @droidmonkey On Ubuntu 22.04 I have the same error with 2.7.1. I tried with wayland and X11. This is the error:

```console
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockContainer>[<Gjs_ubuntu-dock_ubuntu_com_docking_DashToDock>:0x558087e45190] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<Gjs_ubuntu-dock_ubuntu_com_docking_DashSlideContainer>:0x5580868018c0] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockBox>[<StBoxLayout>:0x558085823a20] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dash>[<Gjs_ubuntu-dock_ubuntu_com_dash_DockDash>:0x558087402420] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<StWidget>:0x55808a421c20] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<ClutterActor>:0x558085edd940] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockDashContainer>[<StBoxLayout>:0x558087e34eb0] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockDashScrollview>[<StScrollView>:0x55808a48cb00] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<StBoxLayout>:0x558087a94450] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockContainer>[<Gjs_ubuntu-dock_ubuntu_com_docking_DashToDock>:0x558087e45190] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<Gjs_ubuntu-dock_ubuntu_com_docking_DashSlideContainer>:0x5580868018c0] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockBox>[<StBoxLayout>:0x558085823a20] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dash>[<Gjs_ubuntu-dock_ubuntu_com_dash_DockDash>:0x558087402420] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<StWidget>:0x55808a421c20] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<ClutterActor>:0x558085edd940] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockDashContainer>[<StBoxLayout>:0x558087e34eb0] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <dashtodockDashScrollview>[<StScrollView>:0x55808a48cb00] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 gnome-shell[1703]: Can't update stage views actor <unnamed>[<StBoxLayout>:0x558087a94450] is on because it needs an allocation.
Apr 12 21:54:02 bastien-ThinkPad-X260 systemd[1489]: snap.keepassxc.keepassxc.f2cf3fbd-a575-4f32-b6a0-ce6ad9218eea.scope: Consumed 1.404s CPU time.
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Activating service name='org.gnome.ControlCenter.SearchProvider' requested by ':1.38' (uid=1000 pid=1703 comm="/usr/bin/gnome-shell " label="unconfined")
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Activating service name='org.gnome.Calculator.SearchProvider' requested by ':1.38' (uid=1000 pid=1703 comm="/usr/bin/gnome-shell " label="unconfined")
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Activating service name='org.gnome.Characters.BackgroundService' requested by ':1.38' (uid=1000 pid=1703 comm="/usr/bin/gnome-shell " label="unconfined")
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Successfully activated service 'org.gnome.ControlCenter.SearchProvider'
Apr 12 21:54:05 bastien-ThinkPad-X260 nautilus[3559]: Source ID 28674 was not found when attempting to remove it
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Successfully activated service 'org.gnome.Characters.BackgroundService'
Apr 12 21:54:05 bastien-ThinkPad-X260 dbus-daemon[1512]: [session uid=1000 pid=1512] Successfully activated service 'org.gnome.Calculator.SearchProvider'
Apr 12 21:54:05 bastien-ThinkPad-X260 nautilus[3559]: Source ID 28685 was not found when attempting to remove it
Apr 12 21:54:05 bastien-ThinkPad-X260 systemd[1489]: Started Application launched by gnome-shell.
Apr 12 21:54:05 bastien-ThinkPad-X260 systemd[1489]: Started snap.keepassxc.keepassxc.1aa1b2b0-abc9-4249-833c-a2dc050417d3.scope.
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.370638] kauditd_printk_skb: 19 callbacks suppressed
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.370641] audit: type=1400 audit(1649793245.794:1034): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libdl-2.31.so" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.370714] audit: type=1400 audit(1649793245.794:1035): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.371150] audit: type=1400 audit(1649793245.794:1036): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libdbus-1.so.3.19.11" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.371475] audit: type=1400 audit(1649793245.794:1037): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libsystemd.so.0.28.0" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.371852] audit: type=1400 audit(1649793245.798:1038): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libzstd.so.1.4.4" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.371987] audit: type=1400 audit(1649793245.798:1039): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.6400.6" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.372180] audit: type=1400 audit(1649793245.798:1040): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libkrb5.so.3.3" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.372302] audit: type=1400 audit(1649793245.798:1041): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.372373] audit: type=1400 audit(1649793245.798:1042): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libcom_err.so.2.1" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 kernel: [ 4842.372474] audit: type=1400 audit(1649793245.798:1043): apparmor="DENIED" operation="open" profile="snap.keepassxc.keepassxc" name="/snap/core20/1405/usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1" pid=11831 comm="keepassxc" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Apr 12 21:54:05 bastien-ThinkPad-X260 keepassxc[11831]: Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Apr 12 21:54:05 bastien-ThinkPad-X260 keepassxc[11831]: Qt: Session management error: Could not open network socket
```

**Comment by droidmonkey** [comment]
I cannot provide support for that problem. I ran through snap on 22.04 yesterday without issue. Reinstall snapd and the snap.

**Comment by washiur17** [comment]
In my Ubuntu 20.04 - the issue is solved and working properly with 2.7.1. Thanks for this version.
